### ISSUE TITLE
Storage Account type UltraSSD_LRS is not supported with caching type ReadWrite in CSI migration scenario

### ISSUE THREAD

**Issue by axelgMS** [kubernetes-sigs/azuredisk-csi-driver#1131]
**What happened**:

CRI 281859103 - Following steps on https://docs.microsoft.com/en-us/azure/aks/use-ultra-disks, and getting error below:

  _"error": {
    "code": "InvalidParameter",
    "message": "Storage Account type UltraSSD_LRS is not supported with caching type ReadWrite. Please use caching type None.",
    "target": "dataDisk.caching"_

**What you expected to happen**:

no error

**How to reproduce it**:

Follow the steps on https://docs.microsoft.com/en-us/azure/aks/use-ultra-disks

**Anything else we need to know?**:

using the Storage Class below works fine:

_kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: ultra-disk-sc
**provisioner: disk.csi.azure.com**
volumeBindingMode: WaitForFirstConsumer # optional, but recommended if you want to wait until the pod that will use this disk is created 
parameters:
  skuname: UltraSSD_LRS
  kind: managed
  cachingmode: None
  diskIopsReadWrite: "2000"  # minimum value: 2 IOPS/GiB 
  diskMbpsReadWrite: "320"   # minimum value: 0.032/GiB_

**Environment**:
- CSI Driver version:  1.8.0 ?
- Kubernetes version (use `kubectl version`): AKS - k8s 1.21.7
- OS (e.g. from /etc/os-release):  Ubuntu 18.04
- Kernel (e.g. `uname -a`): 
- Install tools:
- Others:

**Comment by andyzhangx** [comment]
I have worked out a aks doc PR to fix the doc issue first, we don't want users to still use in-tree storage class on aks 1.21+, so the fix on csi migration would be in low priority.

```console
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: ultra-disk-sc
provisioner: disk.csi.azure.com # replace with "kubernetes.io/azure-disk" if aks version is below than 1.21
volumeBindingMode: WaitForFirstConsumer # optional, but recommended if you want to wait until the pod that will use this disk is created 
parameters:
  skuname: UltraSSD_LRS
  kind: managed
  cachingMode: None
  diskIopsReadWrite: "2000"  # minimum value: 2 IOPS/GiB 
  diskMbpsReadWrite: "320"   # minimum value: 0.032/GiB
```

**Comment by andyzhangx** [comment]
the original issue was fixed in 1.22.6, 1.23.2
